primitives/src/pool.rs

```diff
@@ -55,8 +57,8 @@ where
 
 impl<Balance, MarketId> MaxEncodedLen for Pool<Balance, MarketId>
 where
-    Balance: MaxEncodedLen,
-    MarketId: MaxEncodedLen,
+    Balance: MaxEncodedLen + Decode,
+    MarketId: MaxEncodedLen + Decode,
```

Comment:
It's not necessary to require that `Balance` and `MarketId` implement `Decode` (here and further below). When deriving `Decode`, the compiler implements `Decode` if all generic parameters satisfy the constraint `Decode`. In particular, it's not necessary for the functions in `arbitrage.rs` to require that trait bound.